GPU CI is currently unavailable

## 🐛 Bug
Since switching to our cluster in https://github.com/Lightning-AI/lightning-bolts/pull/913, we started to see the following errors stochastically:

```
/usr/bin/docker create --name 85878207657742a685e5af4e546e256b_pytorchlightningpytorch_lightningbasecudapy39torch112cuda1161_97286f --label 83d4d6 --network vsts_network_c70902dd7f974160b8eed6fb132270b8 --gpus=all --shm-size=16g --name ci-container -v /usr/bin/docker:/tmp/docker:ro -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/azp/agent-litGPU-19_2,3/_work/5":"/__w/5" -v "/azp/agent-litGPU-19_2,3/_work/_temp":"/__w/_temp" -v "/azp/agent-litGPU-19_2,3/_work/_tasks":"/__w/_tasks" -v "/azp/agent-litGPU-19_2,3/_work/_tool":"/__t" -v "/azp/agent-litGPU-19_2,3/externals":"/__a/externals":ro -v "/azp/agent-litGPU-19_2,3/_work/.taskkey":"/__w/.taskkey" pytorchlightning/pytorch_lightning:base-cuda-py3.9-torch1.12-cuda11.6.1 "/__a/externals/node/bin/node" -e "setInterval(function(){}, 24 * 60 * 60 * 1000);"
Error response from daemon: Conflict. The container name "/ci-container" is already in use by container "b22e06b712c25f35c3366c7d14130f55b09b510caff2c8591dbd1f4eecce6dd6". You have to remove (or rename) that container to be able to reuse that name.
##[error]Exit code 1 returned from process: file name '/usr/bin/docker', arguments 'create --name 85878207657742a685e5af4e546e256b_pytorchlightningpytorch_lightningbasecudapy39torch112cuda1161_97286f --label 83d4d6 --network vsts_network_c70902dd7f974160b8eed6fb132270b8 --gpus=all --shm-size=16g --name ci-container -v /usr/bin/docker:/tmp/docker:ro -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/azp/agent-litGPU-19_2,3/_work/5":"/__w/5" -v "/azp/agent-litGPU-19_2,3/_work/_temp":"/__w/_temp" -v "/azp/agent-litGPU-19_2,3/_work/_tasks":"/__w/_tasks" -v "/azp/agent-litGPU-19_2,3/_work/_tool":"/__t" -v "/azp/agent-litGPU-19_2,3/externals":"/__a/externals":ro -v "/azp/agent-litGPU-19_2,3/_work/.taskkey":"/__w/.taskkey" pytorchlightning/pytorch_lightning:base-cuda-py3.9-torch1.12-cuda11.6.1 "/__a/externals/node/bin/node" -e "setInterval(function(){}, 24 * 60 * 60 * 1000);"'.
```

### To Reproduce
- https://dev.azure.com/Lightning-AI/lightning%20Bolts/_build/results?buildId=110442&view=logs&j=cb44143e-1b72-550b-97cf-26ffc4f8d032
```
Agent name: 'litGPU-19_2,3'
Agent machine name: 'lambda-server5'
Current agent version: '2.211.0'
```
- https://dev.azure.com/Lightning-AI/lightning%20Bolts/_build/results?buildId=110437&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29&t=cfe97727-512b-4cd5-ab8b-4a53465e89f8
```
Agent name: 'litGPU-21_6,7'
Agent machine name: 'lambda-server3'
Current agent version: '2.211.0'
```

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -21,11 +21,12 @@ jobs:
     pool: lit-rtx-3090
     variables:
       DEVICES: $( python -c 'print("$(Agent.Name)".split("_")[-1])' )
+      CONTAINER_NAME: $( echo $RANDOM )
 
     container:
       image: "pytorchlightning/pytorch_lightning:base-cuda-py3.9-torch1.12-cuda11.6.1"
       #endpoint: azureContainerRegistryConnection
-      options: "--gpus=all --shm-size=16g --name ci-container -v /usr/bin/docker:/tmp/docker:ro"
+      options: "--gpus=all --shm-size=16g --name $(CONTAINER_NAME) -v /usr/bin/docker:/tmp/docker:ro"
 
     workspace:
       clean: all
@@ -50,7 +51,7 @@ jobs:
       displayName: 'Sanity check'
 
     - script: |
-        /tmp/docker exec -t -u 0 ci-container \
+        /tmp/docker exec -t -u 0 $(CONTAINER_NAME) \
         sh -c "apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confold" -y install sudo"
       displayName: 'Install Sudo in container (thanks Microsoft!)'
 

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -35,18 +35,16 @@ jobs:
     - bash: |
         echo "##vso[task.setvariable variable=CUDA_VISIBLE_DEVICES]$(DEVICES)"
         echo "##vso[task.setvariable variable=CID]$(head -1 /proc/self/cgroup|cut -d/ -f3)"
-      displayName: 'set visible devices'
+      displayName: 'Set environment variables'
 
     - bash: |
         whereis nvidia
         nvidia-smi
         echo $CUDA_VISIBLE_DEVICES
+        echo $CID
         python --version
         pip --version
         pip list
-        echo $CID
-        cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’
-        head -1 /proc/self/cgroup|cut -d/ -f3
       displayName: 'Image info & NVIDIA'
 
     - bash: |

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -45,6 +45,7 @@ jobs:
         pip list
         export CID=$(cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’)
         echo $CID
+        cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’
       displayName: 'Image info & NVIDIA'
 
     - bash: |

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -40,10 +40,11 @@ jobs:
         whereis nvidia
         nvidia-smi
         echo $CUDA_VISIBLE_DEVICES
-        echo $CONTAINER_NAME
         python --version
         pip --version
         pip list
+        export CID=$(cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’)
+        echo $CID
       displayName: 'Image info & NVIDIA'
 
     - bash: |
@@ -51,7 +52,7 @@ jobs:
       displayName: 'Sanity check'
 
     - script: |
-        /tmp/docker exec -t -u 0  \
+        /tmp/docker exec -t -u 0 $CID \
         sh -c "apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confold" -y install sudo"
       displayName: 'Install Sudo in container (thanks Microsoft!)'
 

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -11,9 +11,6 @@ pr:
   - master
   - release/*
 
-variables:
-  CONTAINER_NAME: $( echo $RANDOM )
-
 jobs:
   - job: pytest
     # how long to run the job before automatically cancelling
@@ -28,7 +25,7 @@ jobs:
     container:
       image: "pytorchlightning/pytorch_lightning:base-cuda-py3.9-torch1.12-cuda11.6.1"
       #endpoint: azureContainerRegistryConnection
-      options: "--gpus=all --shm-size=16g --name $(CONTAINER_NAME) -v /usr/bin/docker:/tmp/docker:ro"
+      options: "--gpus=all --shm-size=16g -v /usr/bin/docker:/tmp/docker:ro"
 
     workspace:
       clean: all
@@ -54,7 +51,7 @@ jobs:
       displayName: 'Sanity check'
 
     - script: |
-        /tmp/docker exec -t -u 0 $(CONTAINER_NAME) \
+        /tmp/docker exec -t -u 0  \
         sh -c "apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confold" -y install sudo"
       displayName: 'Install Sudo in container (thanks Microsoft!)'
 

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -34,14 +34,14 @@ jobs:
 
     - bash: |
         echo "##vso[task.setvariable variable=CUDA_VISIBLE_DEVICES]$(DEVICES)"
-        echo "##vso[task.setvariable variable=CID]$(head -1 /proc/self/cgroup|cut -d/ -f3)"
+        echo "##vso[task.setvariable variable=CONTAINER_ID]$(head -1 /proc/self/cgroup|cut -d/ -f3)"
       displayName: 'Set environment variables'
 
     - bash: |
         whereis nvidia
         nvidia-smi
         echo $CUDA_VISIBLE_DEVICES
-        echo $CID
+        echo $CONTAINER_ID
         python --version
         pip --version
         pip list
@@ -52,7 +52,7 @@ jobs:
       displayName: 'Sanity check'
 
     - script: |
-        /tmp/docker exec -t -u 0 $CID \
+        /tmp/docker exec -t -u 0 $CONTAINER_ID \
         sh -c "apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confold" -y install sudo"
       displayName: 'Install Sudo in container (thanks Microsoft!)'
 

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -34,6 +34,7 @@ jobs:
 
     - bash: |
         echo "##vso[task.setvariable variable=CUDA_VISIBLE_DEVICES]$(DEVICES)"
+        echo "##vso[task.setvariable variable=CID]$(cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’)"
       displayName: 'set visible devices'
 
     - bash: |
@@ -43,7 +44,6 @@ jobs:
         python --version
         pip --version
         pip list
-        export CID=$(cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’)
         echo $CID
         cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’
       displayName: 'Image info & NVIDIA'

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -11,6 +11,9 @@ pr:
   - master
   - release/*
 
+variables:
+  CONTAINER_NAME: $( echo $RANDOM )
+
 jobs:
   - job: pytest
     # how long to run the job before automatically cancelling
@@ -21,7 +24,6 @@ jobs:
     pool: lit-rtx-3090
     variables:
       DEVICES: $( python -c 'print("$(Agent.Name)".split("_")[-1])' )
-      CONTAINER_NAME: $( echo $RANDOM )
 
     container:
       image: "pytorchlightning/pytorch_lightning:base-cuda-py3.9-torch1.12-cuda11.6.1"
@@ -41,6 +43,7 @@ jobs:
         whereis nvidia
         nvidia-smi
         echo $CUDA_VISIBLE_DEVICES
+        echo $CONTAINER_NAME
         python --version
         pip --version
         pip list

--- a/.azure/gpu-tests.yml
+++ b/.azure/gpu-tests.yml
@@ -34,7 +34,7 @@ jobs:
 
     - bash: |
         echo "##vso[task.setvariable variable=CUDA_VISIBLE_DEVICES]$(DEVICES)"
-        echo "##vso[task.setvariable variable=CID]$(cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’)"
+        echo "##vso[task.setvariable variable=CID]$(head -1 /proc/self/cgroup|cut -d/ -f3)"
       displayName: 'set visible devices'
 
     - bash: |
@@ -46,6 +46,7 @@ jobs:
         pip list
         echo $CID
         cat /proc/self/cgroup | head -1 | tr --delete ‘10:memory:/docker/’
+        head -1 /proc/self/cgroup|cut -d/ -f3
       displayName: 'Image info & NVIDIA'
 
     - bash: |

